core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -561,6 +561,26 @@ sealed abstract class Parser[+A] extends Parser0[A] {
   def repSep(min: Int, max: Int, sep: Parser0[Any]): Parser[NonEmptyList[A]] =
     Parser.repSep(this, min = min, max = max, sep = sep)
 
+  /** Repeat this parser 0 or more times until `end` Parser succeeds.
+    */
+  def repUntil0(end: Parser[Any]): Parser0[List[A]] =
```

Comment:
One minor issue: I think end can always be a Parser0 without sacrificing any safety.

For instance it could be a Parser.end which could be potentially useful or it could be a parser that would expects the index to be in a certain range.

Reply:
Yea, I see, that does make sense. Let me update it.